### DISCUSSION TITLE
Razzle/Vercel(aka zeit) deployement

### DISCUSSION THREAD

**Discussion by momosetti** [jaredpalmer/razzle#1355] [Q&A] [answered]
## ❓Question
Any deployment  guide for deploy razzle app on vercel? thanks!

**Comment by whjvenyl** [comment, marked as answer]
Create this vercel.json in the root directory of your project:
```javascript
{
    "name": "after-js",
    "version": 2,
    "builds": [
        {
            "src": "build/public/**",
            "use": "@now/static"
        },
        {
            "src": "build/server.js",
            "use": "@now/node-server"
        }
    ],
    "routes": [
        { "src": "/assets.json", "dest": "build/assets.json" },
        { "src": "/chunks.json", "dest": "build/chunks.json" },
        { "src": "/favicon.ico", "dest": "build/public/favicon.ico" },
        { "src": "/robots.txt", "dest": "build/public/robots.txt" },
        { "src": "/static/(.*)", "dest": "build/public/static/$1" },
        { "src": "/(.*)", "dest": "build/server.js" }
        
    ]
}
```

Then do a local build with 'npm run build'. 
Afterwards deploy from a local terminal with 'vercel'

**Reply by nimaa77** [reply]
thanks

and you can remove these two lines too (they don't get used at run time, it's just a build time thing):
```json
        { "src": "/assets.json", "dest": "build/assets.json" },
        { "src": "/chunks.json", "dest": "build/chunks.json" },
```

**Reply by olup** [reply]
And is there a way to make it work with git repo auto linking ? as in - i don't build on my computer ? Trying to get it to work but i doesn't and vercel is not building the project as there is a "builds" configuration in the json. Any clues ?

**Reply by momosetti** [reply]
thanks @whjvenyl .

**Reply by sickdyd** [reply]
Is there any solution to this issue yet? To make it work with vercel I had to remove the `build` folder from `.gitignore` and push it to my github repo. It seems that is still building on vercel when deploying though. It doesn't seem to make much sense... or am I missing something?

**Reply by fivethreeo** [reply]
Routes is legacy it seems

https://vercel.com/docs/configuration#project/rewrites

https://vercel.com/guides/using-express-with-vercel

https://vercel.com/docs/build-step#build-command

https://vercel.com/docs/runtimes#advanced-usage/advanced-node-js-usage

**Reply by fivethreeo** [reply]
> Is there any solution to this issue yet? To make it work with vercel I had to remove the `build` folder from `.gitignore` and push it to my github repo. It seems that is still building on vercel when deploying though. It doesn't seem to make much sense... or am I missing something?

Rename build in scripts in package.json to something else.

**Reply by moringaman** [reply]
> And is there a way to make it work with git repo auto linking ? as in - i don't build on my computer ? Trying to get it to work but i doesn't and vercel is not building the project as there is a "builds" configuration in the json. Any clues ?

I'm having the same issue, in terms of linking this with the git deployment on vercel whereby vercel is not kicking off the build. Can anyone help out with this??

**Reply by fivethreeo** [reply]
Post your package an vercel json.

**Reply by moringaman** [reply]
Hi,

My package.json is as follows

```
{
  "name": "chunkysoapui",
  "version": "0.1.0",
  "license": "MIT",
  "scripts": {
    "start": "razzle start",
    "buildscript": "razzle build",
    "test": "razzle test --env=jsdom",
    "start:prod": "NODE_ENV=production node build/server.js"
  },
  "dependencies": {
    "@stripe/react-stripe-js": "^1.1.2",
    "@stripe/stripe-js": "^1.10.0",
    "express": "^4.17.1",
    "fuse.js": "^6.4.6",
    "isomorphic-unfetch": "^3.0.0",
    "jspdf": "^2.3.1",
    "lodash": "^4.17.20",
    "moment": "^2.29.1",
    "path": "^0.12.7",
    "razzle": "^3.1.8",
    "razzle-plugin-pwa": "^2.2.3",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-feather": "^2.0.8",
    "react-loader-spinner": "^3.1.14",
    "react-paginate": "^7.1.2",
    "react-redux": "^7.2.1",
    "react-router-dom": "^5.2.0",
    "redux": "^4.0.5",
    "stripe": "^8.114.0",
    "styled-components": "4.2.0",
    "universal-cookie": "^4.0.4"
  }
}
```

vercel.json is,

```
{
    "version": 2,
    "builds": [
        {
            "src": "build/public/**",
            "use": "@now/static"
        },
        {
            "src": "build/server.js",
            "use": "@now/node-server"
        }
    ],
    "routes": [
        { "src": "/assets.json", "dest": "build/assets.json" },
        { "src": "/chunks.json", "dest": "build/chunks.json" },
        { "src": "/favicon.ico", "dest": "build/public/favicon.ico" },
        { "src": "/robots.txt", "dest": "build/public/robots.txt" },
        { "src": "/static/(.*)", "dest": "build/public/static/$1" },
        { "src": "/(.*)", "dest": "build/server.js" }
        
    ]
}
```

**Reply by fivethreeo** [reply]
Rename buildscript to build

https://vercel.com/docs/configuration#project/rewrites

**Reply by moringaman** [reply]
Hi @fivethreeo, I had it as build initially as per your previous comment as it was not working as build...

**Reply by fivethreeo** [reply]
If you’d like to override the Build Command, you can turn on the Override toggle and specify the command.
